surveys/2022-wg-debugging-survey/questions.md

```diff
@@ -0,0 +1,331 @@
+
+> *Justification:* Cohort analysis
+>
+> Debuggers are often platform-specific and various improvements that the
+> compiler team could make could only benefit a subset of all platforms, so
+> knowing what users are using is valuable.
+
+### How would you rate your Rust expertise?
+Type: select one
+
+- I can't read or write Rust
+- I can write simple programs in Rust
+- I can write useful, production-ready code but it is a struggle
+- I am productive writing Rust
+- I am an expert
+
+> *Justification:* Cohort analysis
+>
+> Users with different levels of experience are likely to face different
+> challenges.
+
+
+### What forms of debugging do you typically use?
+Type: select all that apply
+
+- `println!`/`log`/`tracing`
+- Stack traces
+- Traditional debuggers
+- Record/replay debuggers
+- High-level tracing
+
+> *Justification:*
+>
+> Understand how users are debugging Rust code.
+
+### What types of application are you debugging?
+Type: select all that apply (optional)
+
+- Software
+- Embedded hardware
+- ...
+
+> *Justification:*
+>
+> Some debugger use cases are very different from "standard" debugging.
+
+### Do you debug remotely?
+i.e. are you debugging something that's running on an embedded device/another
+platform, or something like that
+
+Type: yes/no
+
+> *Justification:*
+>
+> Remote debugging will have a different user experience that might be better
+> or worse.
+
+### When do you start debugging?
+Type: select all that apply
+
+- Fixing a crash/panic/access violation
+- Fixing a logic bug
+- Learning how software works
+- Understand runtime semantics
+- Understand a third-party library's behaviour
+- Understand own program's behaviour
+- Profiling
+
+> *Justification:*
+>
+> Different motivations for debugging will benefit from different features from
+> debuggers and might have a better or worse experience.
+
+
+### How would you rate your expertise with debuggers?
+Type: select one
+
+- I have never used a debugger
+- I am uncomfortable using a debugger
+- I am comfortable using a debugger
+- I am very comfortable using a debugger
+
+> *Justification:*
+>
+> Users who are more familiar with debuggers will face different problems than
+> users who are not.
+
+### Why are you uncomfortable using a debugger?
+If: uncomfortable with debugger
+Type: select all that apply
+
+- Unfamiliar (just need to spend more time)
+- Ease-of-use
+- Lack of documentation with Rust
+
+> *Justification:*
+>
+> Why aren't users comfortable with debuggers is useful.
+
+### Do you often use a debugger with other programming languages (anything but Rust)?
+Type: select one
+
+- Daily or nearly so
+- Weekly or nearly so
+- Monthly or nearly so
+- Rarely
+
+> *Justification:*
+>
+> If a user regularly uses a debugger for other languages but not for Rust then
+> it would be interesting to know that.
+
+### Which programming language(s) do you use a debugger with?
+Type: select all that apply
+
+- C
+- C++
+- Java
+- JavaScript
+- Python
+- ...
```

Comment:
Might be worth checking to see if we have survey data about the common languages people use with Rust but I think Go and C# are probably worth being included in this list.

Reply:
Added!